iOS/DuckDuckGo/AIChat/InputBox/SwitchBar/DaxLogoManager.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  DaxLogoManager.swift
+//  DuckDuckGo
+//
+//  Copyright © 2025 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import UIKit
+import SwiftUI
+
+/// Manages the Dax logo view display and positioning
+final class DaxLogoManager {
+    
+    // MARK: - Types
+    
+    struct Configuration {
+        let logoOffset: CGFloat
+        
+        static let `default` = Configuration(logoOffset: 18)
+    }
+    
+    // MARK: - Properties
+    
+    private let configuration: Configuration
+    private var daxLogoHostingController: UIHostingController<NewTabPageDaxLogoView>?
+    private(set) var logoCenterYConstraint: NSLayoutConstraint?
+    
+    var logoView: UIView? {
+        daxLogoHostingController?.view
+    }
+    
+    // MARK: - Initialization
+    
+    init(configuration: Configuration = .default) {
+        self.configuration = configuration
+    }
+    
+    // MARK: - Public Methods
+    
+    /// Installs the Dax logo in the provided parent view controller
+    func installInViewController(_ viewController: UIViewController) {
+        let daxLogoView = NewTabPageDaxLogoView()
+        let hostingController = UIHostingController(rootView: daxLogoView)
+        daxLogoHostingController = hostingController
+        
+        hostingController.view.backgroundColor = .clear
+        viewController.addChild(hostingController)
+        viewController.view.addSubview(hostingController.view)
+        hostingController.view.translatesAutoresizingMaskIntoConstraints = false
+        
+        // Offset so the logo is displayed on the same height as the NTP logo
+        logoCenterYConstraint = hostingController.view.centerYAnchor.constraint(
+            equalTo: viewController.view.centerYAnchor,
+            constant: configuration.logoOffset
+        )
+        
+        NSLayoutConstraint.activate([
+            hostingController.view.centerXAnchor.constraint(equalTo: viewController.view.centerXAnchor),
+            logoCenterYConstraint!
+        ])
+        
+        hostingController.didMove(toParent: viewController)
+        
+        // Send to back so it appears behind other views
+        viewController.view.sendSubviewToBack(hostingController.view)
```

Comment:
Technically this should be before `hostingController.didMove(toParent: viewController)`.